src/main/scala/is/hail/utils/GridPartitioner.scala

```diff
@@ -0,0 +1,21 @@
+package is.hail.utils
+
+import org.apache.spark.Partitioner
+import scala.reflect.classTag
+
+object GridPartitioner {
+
+  def apply(nRowBlocks: Int, nColBlocks: Int): Partitioner = {
+    val intClass = classTag[Int].runtimeClass
+    val gpObjectClass = Class.forName("org.apache.spark.mllib.linalg.distributed.GridPartitioner$")
```

Comment:
This is because it is private?  The GridPartitioner code is pretty short, we might just fork it.